staging/src/k8s.io/code-generator/cmd/informer-gen/generators/factoryinterface.go

```diff
@@ -82,4 +84,16 @@ type SharedInformerFactory interface {
 	Start(stopCh <-chan struct{})
 	InformerFor(obj {{.runtimeObject|raw}}, newFunc NewInformerFunc) {{.cacheSharedIndexInformer|raw}}
 }
+
+type FilterFunc func(*{{.v1ListOptions|raw}}) (namespace string)
```

Comment:
this is a strange mixture of ListOptions tweaking and filtering a namespace. Also mutation in a filter function is kind of strange.

Reply:
Yep - I'm unsure on the best signature for the FilterFunc. I think allowing users to modify ListOptions *somehow* is reasonable (given things like `IncludeUninitialized`).

What do you think of making the argument passed *to* the FilterFunc be a non-pointer, and expecting the user to return a copy of those ListOptions back (with their modifications, and along with the namespace to operate within as already happens)

Reply:
How does the unterlaying store handle uninitialized objects?

Reply:
I've not checked myself, but I'd have thought it'd treat them like any other resource. Upon changes, the store would be updated with the new resource?

From my understanding, uninitialised objects have very little difference to normal objects, except they aren't returned by the apiserver by default (aka without `includeUninitialized=true` on the request) until the `metadata.initializers` field gets set to nil.